test/docker-e2e/dockerchain/testchain.go

```diff
@@ -0,0 +1,145 @@
+package dockerchain
+
+import (
+	"context"
+	"fmt"
+	"github.com/celestiaorg/celestia-app/v5/app"
+	"github.com/celestiaorg/celestia-app/v5/app/encoding"
+	"github.com/celestiaorg/celestia-app/v5/pkg/user"
+	tastoradockertypes "github.com/celestiaorg/tastora/framework/docker"
+	"github.com/celestiaorg/tastora/framework/testutil/config"
+	"github.com/celestiaorg/tastora/framework/testutil/maps"
+	cometcfg "github.com/cometbft/cometbft/config"
+	servercfg "github.com/cosmos/cosmos-sdk/server/config"
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+// NewCelestiaChainBuilder constructs a new ChainBuilder configured for a Celestia instance with predefined parameters.
+func NewCelestiaChainBuilder(t *testing.T, cfg *Config) *tastoradockertypes.ChainBuilder {
+	genesisBz, err := cfg.Genesis.ExportBytes()
+	require.NoError(t, err, "failed to export genesis bytes")
+
+	// TODO: why do I need to do this?
+	// fails with 2025-07-03 14:57:42 Error: failed to get current app state: failed to determine genesis version
+	genesisBz, err = maps.SetField(genesisBz, "consensus.params.version.app", "4")
+	require.NoError(t, err)
+
+	encodingConfig := testutil.MakeTestEncodingConfig(app.ModuleEncodingRegisters...)
+
+	kr := cfg.Genesis.Keyring()
+
+	records, err := kr.List()
+	require.NoError(t, err)
+
+	vals := make([]tastoradockertypes.ChainNodeConfig, len(records))
+	for i, record := range records {
+		val, exists := cfg.Genesis.Validator(i)
+		require.True(t, exists, "validator at index %d should exist", i)
+		privKeyBz, err := val.PrivateKeyBytes()
+		require.NoError(t, err, "failed to get validator private key bytes")
+
+		vals[i] = tastoradockertypes.NewChainNodeConfigBuilder().
+			WithPrivValidatorKey(privKeyBz).
+			WithAccountName(record.Name).
+			WithKeyring(kr).
+			Build()
+	}
+
+	return tastoradockertypes.NewChainBuilder(t).
+		WithName("celestia"). // just influences home directory on the host.
+		WithChainID(cfg.Genesis.ChainID).
+		WithDockerClient(cfg.DockerClient).
+		WithDockerNetworkID(cfg.DockerNetworkID).
+		WithImage(tastoradockertypes.NewDockerImage(cfg.Image, cfg.Tag, "10001:10001")).
+		WithAdditionalStartArgs("--force-no-bbr", "--grpc.enable", "--grpc.address", "0.0.0.0:9090", "--rpc.grpc_laddr=tcp://0.0.0.0:9099").
```

Comment:
[not blocking] Besides `--force-no-bbr` are the other flags configurable via config.toml or app.toml? That _seems_ cleaner.

Reply:
Ahh it seems possible with the `getPostInitModifications` function below

Reply:
yeah good point, we can actually just put anything in the default config override, these were hanging around from before.